src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9201,6 +9206,15 @@ def quantiles_along_axis0(
             ErrorMessage.not_implemented(
                 "quantile is not supported for datetime columns"
             )
+        assert index is None or len(index) == len(
+            q
+        ), f"length of index {index} did not match quantiles {q}"
+        if len(query_compiler.columns) == 1:
```

Comment:
nit: better to use an alternative to `.columns`, I remember it triggered sometimes queries. You can use instead `InternalFrame.data_column_snowflake_quoted_identifiers` or `query_compiler.get_axis_len(axis)`.